### PR TITLE
improve Nx documentation

### PR DESCRIPTION
**Pull request by AlexJuca**
This PR adds several improvements and corrects a few spelling mistakes.

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -71,7 +71,7 @@ defmodule Nx do
         [2, 4, 6]
       >
 
-  In practice, brodcasting is not restricted only to scalars, it
+  In practice, broadcasting is not restricted only to scalars. It
```
Comment (seanmor5) [inline nx/lib/nx.ex:74]:
```suggestion
  In practice, broadcasting is not restricted only to scalars; it
```

I think this might read better

**Review by seanmor5 — CHANGES_REQUESTED** [review]

File: nx/lib/nx.ex (hunk 1 above)
Comment (AlexJuca) [reply nx/lib/nx.ex]: I agree.

**Review by AlexJuca — COMMENTED** [review]
This looks ok.

**Review by seanmor5 — APPROVED** [review]
Thanks!

**Pull request merged**